Sender.py

```diff
@@ -0,0 +1,29 @@
+import random 
+import json
+
+def get_temperature_in_C(min_value,max_value,no_of_samples):
+    return generate_samples(min_value,max_value,no_of_samples)
+
+def celcious_to_farenheit_convertor(temp):
+    return round((temp * 1.8) + 32,2)
+     
+def generate_samples(min_value,max_value,no_of_samples):
+    return random.sample(range(min_value, max_value),no_of_samples)
+  
+def get_charge_rate(min_value,max_value,no_of_samples):
+    return generate_samples(min_value,max_value,no_of_samples)
+
```

Comment:
The name of this function suggests that this is the main function of the sender process. 
However, it doesn't use any of the other functions of the file. Nor is this function itself called anywhere else.

This implies that instead of 'processing data' inside this function, someone else is supposed to process the data. 
While this function is only responsible for dumping that data in a json.